matlab/CMakeLists.txt

```
@@ -17,6 +17,84 @@
 
 cmake_minimum_required(VERSION 3.20)
 
+# Build the Arrow C++ libraries WITHOUT bundled GoogleTest binaries.
+function(BUILD_ARROW)
+  include(ExternalProject)
+
+  set(ARROW_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/arrow_ep-prefix")
+  set(ARROW_INCLUDE_DIR "${ARROW_PREFIX}/include")
+  set(ARROW_LIBRARY_DIR "${ARROW_PREFIX}/lib")
+  set(ARROW_SHARED_LIB "${ARROW_LIBRARY_DIR}/libarrow.so")
+  set(ARROW_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/arrow_ep_build")
+
+  externalproject_add(arrow_ep
+                      SOURCE_DIR "${CMAKE_SOURCE_DIR}/../cpp"
+                      BINARY_DIR "${ARROW_BINARY_DIR}"
+                      CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${ARROW_PREFIX}
```

We need to specify `CMAKE_INSTALL_LIBDIR=lib` explicitly because some platforms such as CentOS uses `lib64` for the default `CMAKE_INSTALL_LIBDIR`:

```suggestion
                      CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${ARROW_PREFIX} -DCMAKE_INSTALL_LIBDIR=lib
```

---

Thanks for pointing this out! I didn't realize that some platforms will install shared libraries to directories other than `lib` by default.